pkg/config/config.go

```diff
@@ -42,6 +42,8 @@ const (
 )
 
 var errorMessages = map[string]string{
+	// general
+	"appendToNonList": "cannot append to a setting whose type is not a list",
```

Comment:
Is there a reason to use lookup-map for error messages? There are few problems with this approach:
- Unused entries can go unnoticed;
- Missing entry can cause runtime error;
- Extra few characters of code, worse readability.

Can we use string constants instead?

Reply:
I think the intent is to eventually simplify the range of error messages and fail with more custom error types that implement the error interface (`func Error() string`). In particular, I do not like the unit tests "grepping" for information.

For example:

https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/dda341f3e8af56790ca59323364e9519d79448ec/pkg/config/config.go#L686-L693

If you'd like to capture that as a task, it would be a good one. I think it ought to have a small design doc with a well documented approach and example PR replacing one or more of the existing messages.

Reply:
> I think the intent is to eventually simplify the range of error messages and fail with more custom error types that implement the error interface 
>  I do not like the unit tests "grepping" for information.

I think is this is orthogonal to what I've meant. The only change I suggest is to not add entry to this map, instead declare `const errMsgAppendToNonList string = "....."` and use it. 

>  I think it ought to have a small design doc with a well documented approach

I think this changes is too trivial for a formal design doc.

I understand that you may hesitate to disrupt the "local style" though it's problematic. Feel free to resolve this comment.

Reply:
I agree with @mr0re1, I believe on all accounts. 

I would not require it as part of this PR as I do not want to discourage the good samaritan cleanup that this PR represents. 

@mr0re1, Do you suggest @tpdownes use this pattern only for the new variable added, or clean up all previous usage in this file?

Reply:
> Do you suggest @tpdownes use this pattern only for the new variable added, or clean up all previous usage in this file?

Only the new message. But, @tpdownes , feel free to not do it.

Reply:
My strong preference is to create a task that defines the appropriate cleanup approach for all `errorMessages` and then another task that does it. I agree that this approach is not desirable, for the reasons stated and I am sure that there are more.